File: Array/beautiful_arrangement_2/julia/beautiful_arrangement_2.jl
# command line input
# n k
n = parse(Int64,ARGS[1])
k = parse(Int64,ARGS[2])
if k>=n
	println([])
else
	temp = collect(1:n)
	tempSlice = collect(1:0)
	for i=1:round(Int64,k/2)+1
		if temp[i]==temp[k-i+2]
			push!(tempSlice,temp[i])
		end
		push!(tempSlice,temp[i],temp[k-i+2])
	end
	for i=1:(k+1)
		temp[i]=tempSlice[i]
	end
	println(temp)
end
	

	

File: Array/beautiful_arrangement_2/dart/beautiful_arrangement_2.dart
import 'dart:io';

// command line input
// n k
main(List<String> ARGS) {
  print(arrangements(int.parse(ARGS[0]), int.parse(ARGS[1])));
}

List<int> arrangements(int n, int k) {
  if (k >= n) {
    return [];
  }
  var temp = List<int>.filled(n, 0);
  for (int i = 0; i < n; i++) {
    temp[i] = i + 1;
  }
  var tempSlice = List<int>.filled(0, 0, growable: true);
  for (int i = 0; i < k / 2 + 1; i++) {
    if (temp[i] == temp[k - i]) {
      tempSlice.add(temp[i]);
    }
    tempSlice.add(temp[i]);
    tempSlice.add(temp[k - i]);
  }
  for (int i = 0; i < k + 1; i++) {
    temp[i] = tempSlice[i];
  }
  return temp;
}
